File: src/test/java/pl/training/module3/adapter/InMemoryQuoteRepositoryTest.java
package pl.training.module3.adapter;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.math.BigDecimal;
import java.util.List;

import org.junit.jupiter.api.Test;

import pl.training.module3.domain.DeliveryQuote;
import pl.training.module3.domain.Parcel;
import pl.training.module3.domain.ShippingMethod;

final class InMemoryQuoteRepositoryTest {
    @Test
    void storesQuote() {
        InMemoryQuoteRepository repository = new InMemoryQuoteRepository();
        DeliveryQuote quote = new DeliveryQuote(
                "developer@example.com",
                ShippingMethod.STANDARD,
                new Parcel(new BigDecimal("3.00")),
                new BigDecimal("17.28"));

        repository.save(quote);

        assertEquals(List.of(quote), repository.quotes());
    }
}

File: csharp/tests/Training.Module3.Tests/Adapter/InMemoryQuoteRepositoryTest.cs
using Training.Module3.Adapter;
using Training.Module3.Domain;

namespace Training.Module3.Tests.Adapter;

public sealed class InMemoryQuoteRepositoryTest
{
    [Fact]
    public void StoresQuote()
    {
        InMemoryQuoteRepository repository = new();
        DeliveryQuote quote = new(
            "developer@example.com",
            ShippingMethod.Standard,
            new Parcel(3.00m),
            17.28m);

        repository.Save(quote);

        Assert.Equal([quote], repository.Quotes());
    }
}
